CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,497 @@
+
+Changes typically include improving code style, tweaking phrasing and quest settings.
+
+You are also welcome to help with reviewing other PRs - different people have different strengths, there are active reviewers who help with code style, there are some native speakers of English, some with deep knowledge of OSM tagging schemes or deep knowledge of how StreetComplete works.
+
+In case that pull requests appears to be ready, it will be marked as approved and wait for merge.
+
+After the PR is finished it will be merged before the beta release of the next version. This way it can be additionally tested with a wider audience before release to all and translators have time to [translate text into other languages](CONTRIBUTING.md#translating-the-app).
+
+After full release it will reach the entire StreetComplete audience who now will be able to more easily contribute to OpenStreetMap. Thanks in advance for that!
+
+# Bad documentation is a bug
+
+Unclear documents, including this one, are a bug. Feel free to either submit a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) with a fix or [open an issue](https://github.com/streetcomplete/StreetComplete/issues/new) describing your confusion.
+
+Note that not everything will be directly described. This document very intentionally doesn't include a step-by-step guide to installing Android Studio, [linking](CONTRIBUTING.md#development) to official docs instead.
+
+# More complexity
+What was described above is an attempt to cover all aspects of quest creation, without describing all the complexity.
+
+Below is some additional info.
+
+## getTitle and getTitleArgs
+
+In some cases it is desirable to change quest title, based on specific object.
+
+This requires preparing space in the message for filling at runtime, and to add the mechanism supplying data to replace placeholders.
+
+### Display value of tag if it is present
+
+[AddFireHydrantDiameter](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/fire_hydrant_diameter/AddFireHydrantDiameter.kt) displays `ref` code of hydrant, if tagged.
+
+```
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> =
+        arrayOfNotNull(tags["ref"])
+
+    override fun getTitle(tags: Map<String, String>): Int {
+        val hasRef = tags.containsAnyKey("ref")
+        return when {
+            hasRef -> R.string.quest_fireHydrant_diameter_ref_title
+            else   -> R.string.quest_fireHydrant_diameter_title
+        }
+    }
+```
+
+``getTitleArgs` will return either array with single element containing value of `ref` tag or an empty array. In either cases it will be passed to code displaying title at runtime.
```

Comment:
```suggestion
`getTitleArgs` will return either array with single element containing value of `ref` tag or an empty array. In either cases it will be passed to code displaying title at runtime.
```